File: modules/spring-core/testdata/java/inspection/dependsOn/src/com/DependsOnConfiguration.java
package com;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.DependsOn;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;

interface I {}

@Component
class A implements I {}

@Service
class B implements I {}

@Configuration
public class DependsOnConfiguration {
    @DependsOn("a")
    @Bean I bDependentOnA() { return new B(); }

    @DependsOn("wrongBeanNameAtMethod") // Ultimate handles it wrong. There is no navigation, nor error bean highlight
    @Bean I bDependentOnWrong() { return new B(); }
}

//region Valid
@Component
@DependsOn("a")
class DependentOnABean {}

@Component
@DependsOn({"a", "b", E.CONST_TO_VALID_BEAN})
class DependentOnMultipleBeans {}
//endregion

//region Invalid
@Component
@DependsOn("wrongBeanNameAtClass")
class DependentOnABeanErr {}

@Component
@DependsOn({"wrongBeanNameListedAtClass", "a", E.CONST_TO_NON_EXISTING_BEAN})
class DependentOnMultipleBeansErr {}
//endregion

@Component
class E implements I {
    public static final java.lang.String CONST_TO_VALID_BEAN = "b";
    public static final java.lang.String CONST_TO_NON_EXISTING_BEAN = "nope";
}

File: modules/spring-core/testdata/kotlin/inspection/dependsOn/src/com/DependsOnConfiguration.kt
package com

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.DependsOn
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service

interface I

@Component
internal class A : I

@Service
internal class B : I

@Configuration
open class DependsOnConfiguration {
    @DependsOn("a")
    @Bean
    open fun bDependentOnA(): I {
        return B()
    }

    @DependsOn("wrongBeanNameAtMethod") // Ultimate handles it wrong. There is no navigation, nor error bean highlight
    @Bean
    open fun bDependentOnWrong(): I {
        return B()
    }
}

//region Valid
@Component
@DependsOn("a")
internal class DependentOnABean

@Component
@DependsOn("a", "b", E.CONST_TO_VALID_BEAN)
internal class DependentOnMultipleBeans

//endregion
//region Invalid
@Component
@DependsOn("wrongBeanNameAtClass")
internal class DependentOnABeanErr

@Component
@DependsOn("wrongBeanNameListedAtClass", "a", E.CONST_TO_NON_EXISTING_BEAN)
internal class DependentOnMultipleBeansErr

//endregion
@Component
internal object E : I {
    const val CONST_TO_VALID_BEAN: String = "b"
    const val CONST_TO_NON_EXISTING_BEAN: String = "nope"
}